### PR TITLE
Add suite content support for test reports

### PR DESCRIPTION
**Pull request by JasperCraeghs**
- Add RST content for test suites via .yml.mako templates
- Support two modes: itemized suites and header content
- When using --itemize-suites: content added to individual suite items
- When not itemizing: content appears above .. contents:: section
- Child suites can access parent suite content files
- Comprehensive test coverage and documentation
- Maintains full backward compatibility"

### CODE REVIEW

File: tests/acceptance_test.py [hunk 1]
```diff
@@ -325,6 +325,31 @@ def test_xunit_nested_suites(self):
...
+        rst_file_name = '{}_report.rst'.format(file_name)
+        xml_file_name = '{}.xml'.format(file_name)
+        input_xml = str(TEST_IN_DIR / xml_file_name)
+        output_rst = str(TEST_OUT_DIR / rst_file_name)
+        xunit2rst_check(input_xml, output_rst, itemize_suites=False, prefix='UTEST_MY_LIB-')
+
+        reference_rst = str(TEST_IN_DIR / rst_file_name)
+        assert filecmp.cmp(output_rst, reference_rst)
+
+
```
Comment (JokeWaumans) [inline tests/acceptance_test.py:333]:
```suggestion
```

File: tests/mako_test.py [hunk 2]
```diff
@@ -27,9 +27,8 @@ def test_mako_error_handling(self):
                            log_cm.output[0])
         test_case.assertIn('File ', log_cm.output[-1])
         test_case.assertIn('line ', log_cm.output[-1])
-        test_case.assertIn("in render_body: '% for suite_idx, suite in enumerate(test_suites):'",
+        test_case.assertIn("for suite_idx in indexed_extra_content_map:",
                            log_cm.output[-1])
```
Comment (JokeWaumans) [inline tests/mako_test.py:32]:
```suggestion


```

File: README.rst [hunk 3]
```diff
@@ -197,6 +212,147 @@ Inside your Mako template, you have access to the following variables, in additi
...
+    UTEST_MY_LIB-SUITE_WITH_CONTENT: |
+      .. note::
+          This content applies to the entire test suite.
+
+          All test cases in this suite are related to testing
+          the core functionality of MY_LIB.
+
+When processed with ``mlx.xunit2rst -i input.xml -o output.rst -s``, this will generate a traceability item
+for the test suite that includes the extra content.
+
```
Comment (JokeWaumans) [inline README.rst:204]: (OPTIONAL) Maybe add `Generated RST output structure:` here as well

File: README.rst [hunk 4]
```diff
@@ -197,6 +212,147 @@ Inside your Mako template, you have access to the following variables, in additi
...
+.. code:: yaml
+
+    UTEST_MY_LIB-CORE_FUNCTIONS: |
+      .. note::
+          **Test Suite Overview**
+
+          This suite tests the core functionality of MY_LIB.
+          All functions are critical for system operation.
+
+Generated RST output structure:
```
Comment (JokeWaumans) [inline README.rst:236]: (OPTIONAL) Add the command used (just to make the structure the same as `Test Suite Content Example`)

File: README.rst [hunk 5]
```diff
@@ -197,6 +212,147 @@ Inside your Mako template, you have access to the following variables, in additi
...
+      **Core Functions Suite**
+
+      Tests fundamental operations of MY_LIB.
+
+    UTEST_MY_LIB-EDGE_CASES: |
+      **Edge Cases Suite**
+
+      Tests boundary conditions and error scenarios.
+
+**With** ``--itemize-suites``:
```
Comment (JokeWaumans) [inline README.rst]: (OPTIONAL) Same here? Add the actual output with the command?

File: README.rst [hunk 6]
```diff
@@ -168,6 +168,21 @@ mapping, aka `dictionary`_, of case insensitive test case names (or test suite n
...
+
+**Suite Content Behavior:**
+
+- **With** ``--itemize-suites``: Suite content is added to individual traceability items for each itemized test suite
+- **Without** ``--itemize-suites``: Suite content is added as header content above the ``.. contents::`` section
+
+**Parent-Child Suite Relationships:**
+
+When test suites have a hierarchical structure (parent suites containing child suites), only child suites with actual test cases get itemized when using ``--itemize-suites``. Parent suites (typically with ``tests="0"``) are not itemized, but their content files are still processed. Child suites can access content defined for both parent and child suite names in the same content file.
+
```
Comment (JokeWaumans) [inline README.rst]: Consider adding a small note about what happens if both parent and child suites define content for the same suite key (which one takes precedence).

**Review by JokeWaumans — APPROVED** [review]

File: README.rst (hunk 3 above)
Comment (JasperCraeghs) [reply README.rst:204]: the user should run it and can see what it looks like imo

File: README.rst (hunk 6 above)
Comment (JasperCraeghs) [reply README.rst]: This is not a thing. The documentation was wrong. The feature is very basic actually. I fixed the readme

File: README.rst (hunk 5 above)
Comment (JasperCraeghs) [reply README.rst]: No need to make this distinction imo, after simplification of implemnetation
